### PR TITLE
Runner module

### PR DESCRIPTION
**Pull request by toffernator**
## Describe your changes

Created `runner.py` which is responsible for taking a quality model, executing its metrics and aggregations functions, and producing a .csv file. `runner.py` includes the date/time down to the second as the tile of the .csv file and adds the name of the quality model used to produce the values in the .csv file.

We should merge #34 first, and use that here to skip the qualitymodel=<qualtiymodel> line that is now written to csv.

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [x] Have I requested and notified the boys for review of the pull-request?

### CODE REVIEW

**Review by FrederikRothe — COMMENTED** [review]
A couple of clarifying comments on the code, maybe just me being slow 🐌

File: models/code_climate.py [hunk 1]
```diff
@@ -29,67 +29,36 @@ def getDesc(self) -> Dict:
...
-                if row["Metric"] == "return_statements":
-                    return int(row["Value"])
+    def structural_issues(self, results: Dict):
+        return (
+            results["argument_count"] + results["file_length"] + results["method_count"]
+        )
 
-    def similar_blocks_of_code(self) -> int | float:
-        return 3
+    def cyclomatic_complexity(self, results: Dict):
```
[the remark this answers is not in the document]
Comment (toffernator) [reply models/code_climate.py:60]: As I understand CodeClimate, they approximate cyclomatic complexity with return statements, and argue that it is good enough when paired with cognitive complexity.

https://docs.codeclimate.com/docs/cyclomatic-complexity#cyclomatic-vs-cognitive-complexity

File: byoqm/runner.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+from byoqm.qualitymodel.qualitymodel import QualityModel
+
+# Assumes the project is being run from the root of the repository
+_OUTPUT_DIR = Path("./output")
+_MODELS_DIR = "models"
+
+
+class Runner:
+    def __init__(self, model_name: str, src_root: str):
```
[the remark this answers is not in the document]
Comment (toffernator) [reply byoqm/runner.py:22]: will update

[the remark this answers is not in the document]
Comment (SadCheeto) [reply byoqm/runner.py]: changed it to a path

File: byoqm/main.py [hunk 3]
```diff
@@ -22,48 +17,15 @@ def parse_src_root() -> Path:
...
-
-
 if __name__ == "__main__":
     if len(sys.argv) < 3:
         print(
             f"Make sure to provide the path to source code and a name of a quality model to use currently have {sys.argv}"
         )
         exit(1)
+    path = Path("byoqm")
+    runner: Runner = Runner("code_climate", path)
```
Comment (Kobo-coder) [inline byoqm/main.py:27]: is this the final decision as to how users changes models, or do we later move this to a parameter for a CLI command? Doesn't stop from merging.

File: byoqm/runner.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+        return results
+
+    def _run_metrics(self) -> Dict:
+        results = {}
+
+        metrics = self._model.getDesc()["metrics"]
+        for metric, exec_path in metrics.items():
+            cmd = [f"./{exec_path}", self._src_root]
+            process = subprocess.run(cmd, stdout=subprocess.PIPE)
```
Comment (Kobo-coder) [inline byoqm/runner.py:81]: Perhaps we should discuss in person, the use of `shell=True` with regards to Helge's recommendation.

File: byoqm/runner.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        current_time: str = time.strftime("%Y-%m-%d_%H-%M-%S", time.gmtime())
+        file_name = Path(current_time + ".csv")
+        file_location = _OUTPUT_DIR / file_name
+
+        with open(file_location, "w") as results_file:
+            writer = csv.writer(results_file)
+            writer.writerow([f"qualitymodel={self._model_name}"])
+            writer.writerow(["metric", "value"])
+            for description, value in results.items():
+                writer.writerow([description, value])
```
Comment (Kobo-coder) [inline byoqm/runner.py:100]: I believe its possible to use pandas to write a csv file as discussed previously. Again, not something that will keep the branch from being merged. Perhaps something we could change later.

**Review by Kobo-coder — APPROVED** [review]
Left some comments. An answer to them before merging would be very appreciated.

File: byoqm/main.py (hunk 3 above)
Comment (SadCheeto) [reply byoqm/main.py:27]: I Think we should discuss this tomorrow. Some of these parameters are optional, but this one should be required

File: byoqm/runner.py (hunk 5 above)
Comment (SadCheeto) [reply byoqm/runner.py:100]: I don't think it is that important as of now, but we could keep it in the backlog

**Pull request merged**